Practicals/Practical3/js/script.js

```diff
@@ -0,0 +1,277 @@
+function onSubmit() {
+   
+  var prodData ;
+
+  if(localStorage.getItem("prodData") == null)
+  {
+    prodData =[];
+  }
+  else{
+    prodData = JSON.parse(localStorage.getItem("prodData"));
+  }
+  if(validateData()){
+    const fileEl = document.getElementById("image");
+ 
+    const fr = new FileReader();
+    fr.readAsDataURL(fileEl.files[0]);
+
+    fr.addEventListener('load', () => {
+      prodData.push({
+        prodId: document.getElementById("prodId").value,
+        name: document.getElementById("name").value,
+        image: fr.result,
+        price: document.getElementById("price").value,
+        description: document.getElementById("description").value,
+      });
+
+      localStorage.setItem("prodData", JSON.stringify(prodData));
+    });
+  }
+}
+
+
+function validateData() {
+  let name = document.getElementById("name").value;
+  let price = document.getElementById("price").value;
+  let id = document.getElementById("prodId").value;
+
+  if (id == "") {
+    document.getElementById("pId").style.display = "block";
```

Comment:
rather than repeating the lines you can create a common function and then use it.